test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/SortTests.cs

```diff
@@ -213,6 +213,164 @@ public async Task GivenQueryWithMultipleFilters_WhenSearchedWithHyphenSortParamO
+        public async Task GivenQueryWithRevinclude_WhenSearchedWithSortParamOnLastupdatedWithHyphen_ThenResourcesAreReturnedInDescendingOrder()
+        {
+            var tag = Guid.NewGuid().ToString();
+
+            List<Resource> resources = new List<Resource>();
+
+            // create the resources which will have an timestamp bigger than the 'now' var
+            Patient[] patients = await CreatePatients(tag);
+
+            List<Observation> observations = new List<Observation>();
+            for (int i = 0; i < patients.Length; i++)
+            {
+                var obs = await AddObservationToPatient(patients[i], "1990-01-01", tag);
+                observations.Add(obs.First());
+            }
+
+            observations.Reverse();
+            resources.AddRange(observations);
+            resources.AddRange(patients.Reverse());
+
+            // Ask to get all patient with specific tag order by birthdate (timestamp)
+            // filter and sort are different based on different types
+            await ExecuteAndValidateBundle($"Patient?_tag={tag}&_sort=-_lastUpdated&_revinclude=Observation:subject", false, resources.ToArray());
+        }
+
+        [Fact]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.SqlServer)]
+        public async Task GivenQueryWithRevinclude_WhenSearchedWithSortParamOnDatetimeWithHyphen_ThenResourcesAreReturnedInDescendingOrder()
+        {
+            var tag = Guid.NewGuid().ToString();
+
+            List<Resource> resources = new List<Resource>();
+
+            // create the resources which will have an timestamp bigger than the 'now' var
+            Patient[] patients = await CreatePatients(tag);
+
+            List<Observation> observations = new List<Observation>();
+            for (int i = 0; i < patients.Length; i++)
+            {
+                var obs = await AddObservationToPatient(patients[i], "1990-01-01", tag);
+                observations.Add(obs.First());
+            }
+
+            resources.AddRange(patients.Reverse());
+            resources.AddRange(observations);
+
+            // Ask to get all patient with specific tag order by birthdate (timestamp)
+            // filter and sort are different based on different types
+            await ExecuteAndValidateBundle($"Patient?_tag={tag}&_sort=-birthdate&_revinclude=Observation:subject", false, resources.ToArray());
+        }
+
+        [Fact]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.SqlServer)]
+        public async Task GivenQueryWithObservationInclude_WhenSearchedWithSortParamOnDatetime_ThenResourcesAreReturnedInAscendingOrder()
+        {
+            var tag = Guid.NewGuid().ToString();
+
+            List<Resource> resources = new List<Resource>();
+
+            // create the resources which will have an timestamp bigger than the 'now' var
+            Patient[] patients = await CreatePatients(tag);
```

Comment:
```suggestion
            var patients = await CreatePatients(tag);
```

and all other places